packages/massa-web3/src/experimental/account.ts

```diff
@@ -0,0 +1,342 @@
+   *
+   * @returns A new address object.
+   */
+  public static fromString(addressBase58Encoded: string): Address {
+    const prefix = addressBase58Encoded.slice(0, ADDRESS_PREFIX.length)
+
+    if (!(prefix == ADDRESS_PREFIX)) {
+      throw new Error(
+        `Invalid address prefix: "${prefix}". The address should start with "${ADDRESS_PREFIX}". Please verify your address and try again.`
+      )
+    }
+
+    let isEOA: boolean
+    switch (addressBase58Encoded[ADDRESS_PREFIX.length]) {
+      case ADDRESS_USER_PREFIX:
+        isEOA = true
+        break
+      case ADDRESS_CONTRACT_PREFIX:
+        isEOA = false
+        break
+      default:
+        throw new Error(
+          `Invalid address prefix: "${addressBase58Encoded[ADDRESS_PREFIX.length]}". 
+          The address should start with "${ADDRESS_PREFIX}U" for EOA or "${ADDRESS_PREFIX}S" for smart contract. 
+          Please verify your address and try again.`
+        )
+    }
+
+    const addressBase58Decoded: Buffer = base58Decode(
+      addressBase58Encoded.slice(ADDRESS_PREFIX.length)
+    )
+    const { value, bytes } = varintDecode(addressBase58Decoded)
+    return new Address(
+      addressBase58Decoded.subarray(bytes),
+      isEOA,
+      value as Version
+    )
+  }
+
+  /**
+   * Initialize a new address object from a raw byte array, a boolean indicating whether the address is an EOA, and a version.
+   * @param bytes - The raw byte array representing the address.
+   * @param isEOA - A boolean indicating whether the address is an EOA.
+   * @param version - The version of the address.
+   *
+   * @returns A new address object.
+   */
+  public static fromBytes(
+    bytes: Uint8Array,
+    isEOA: boolean,
+    version: Version
+  ): Address {
+    return new Address(bytes, isEOA, version)
+  }
+}
+
+export class Signature {
+  public bytes: Uint8Array
+  public version: Version
+
+  private constructor(bytes: Uint8Array, version: Version) {
+    this.bytes = bytes
+    this.version = version
+  }
+
+  /**
+   * Initialize a new signature object from a base58 encoded string.
+   * @param signatureBase58Encoded - The base58 encoded string representing the signature.
+   *
+   * @returns A new signature object.
+   */
+  public static fromString(signatureBase58Encoded: string): Signature {
+    const signatureBase58Decoded: Buffer = base58Decode(signatureBase58Encoded)
+    const { value, bytes } = varintDecode(signatureBase58Decoded)
+    return new Signature(
+      signatureBase58Decoded.subarray(bytes),
+      value as Version
+    )
+  }
+
+  /**
+   * Initialize a new signature object from a raw byte array and a version.
+   * @param bytes - The raw byte array representing the signature.
+   * @param version - The version of the signature.
+   *
+   * @returns A new signature object.
+   */
+  public static fromBytes(bytes: Uint8Array, version: Version): Signature {
+    return new Signature(bytes, version)
+  }
+}
+
+export class Account {
```

Comment:
What about separating Account from Accountv0, Accountv1... ? Accountv0 having a specific configuration in terms of:
- crypto
- string serialisation/deserialisation
- ...